### PR TITLE
Refactor URL Rewrite Logic and Language Handling for Document Processing

### LINKED ISSUE #105: Rewrites do not take storeviews in account
https://github.com/elgentos/magento2-prismicio/blob/dd6f50e7dcb4eec6c4dad86d58cb2917277f4c22/Controller/Webhook/Url.php#L88

https://github.com/elgentos/magento2-prismicio/blob/dd6f50e7dcb4eec6c4dad86d58cb2917277f4c22/Controller/Webhook/Url.php#L145

The store is retrieved through the storeManager, but the webhook is always configured on the default store. So all rewrites are created for the default store.

### PR DESCRIPTION
**Pull request by Ndottens**
Added support for storeviews.

### CODE REVIEW

File: Controller/Webhook/Url.php [hunk 1]
```diff
@@ -103,21 +103,34 @@ public function execute(): ?ResultInterface
...
+            foreach ($stores as $store) {
+                if ($this->configuration->getContentLanguage($store) === $document->lang) {
+                    $currentStore = $store;
+                    break;
+                }
+            }
+
+            if (!$currentStore) {
+                continue;
+            }
```
Comment (michelheitbrink) [inline Controller/Webhook/Url.php]: Same code as in `Controller/Webhook/Unpublished.php`. Way nicer to use an function for it

File: Helper/GetStoreView.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        $this->storeManager = $storeManager;
+        $this->configuration = $configuration;
+    }
+
+    public function getCurrentStoreView(stdClass $document): ?StoreInterface
+    {
+        $stores = $this->storeManager->getStores();
+
+        foreach ($stores as $store) {
+            if ($this->configuration->getContentLanguage($store) === $document->lang) {
```
Comment (michelheitbrink) [inline Helper/GetStoreView.php:29]: What if `lang` doesn't exists? A stdClass is very global

**Review by michelheitbrink — APPROVED** [review]
Added a small comment

File: Helper/GetStoreView.php (hunk 2 above)
Comment (Ndottens) [reply Helper/GetStoreView.php:29]: lang always exist on document.

**Pull request merged**